src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBreakpointEvidenceSpark.java

```diff
@@ -0,0 +1,1160 @@
+ */
+@CommandLineProgramProperties(summary="Find reads that evidence breakpoints.",
+        oneLineSummary="Dump FASTQs for local assembly of putative genomic breakpoints.",
+        programGroup = SparkProgramGroup.class)
+public final class FindBreakpointEvidenceSpark extends GATKSparkTool {
+    private static final long serialVersionUID = 1L;
+    private static final int MIN_MAPQ = 20;
+    private static final int MIN_MATCH_LEN = 45; // minimum length of matched portion of an interesting alignment
+    //private static final int MAX_FRAGMENT_LEN = 2000;
+    private static final int MAX_COVERAGE = 1000;
+    private static final float ASSEMBLY_TO_MAPPED_SIZE_RATIO = 7.f;
+    private static final int MIN_HF_KMER_COUNT = 200;
+
+    private static final SimpleDateFormat dateFormatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss ");
+
+    @Argument(doc = "directory for fastq output", shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME)
+    private String outputDir;
+
+    @Argument(doc = "directory for evidence output", fullName = "breakpointEvidenceDir", optional = true)
+    private String evidenceDir;
+
+    @Argument(doc = "file for breakpoint intervals output", fullName = "breakpointIntervals", optional = true)
+    private String intervalFile;
+
+    @Argument(doc = "file for mapped qname intervals output", fullName = "qnameIntervalsMapped", optional = true)
+    private String qNamesMappedFile;
+
+    @Argument(doc = "file for kmer intervals output", fullName = "kmerIntervals", optional = true)
+    private String kmerFile;
+
+    @Argument(doc = "file for mapped qname intervals output", fullName = "qnameIntervalsForAssembly", optional = true)
+    private String qNamesAssemblyFile;
+
+    /**
+     * This is a path to a file of kmers that appear too frequently in the reference to be usable as probes to localize
+     * reads.  We don't calculate it here, because it depends only on the reference.
+     * The program FindBadGenomicKmersSpark can produce such a list for you.
+     */
+    @Argument(doc = "file containing ubiquitous kmer list", fullName = "kmersToIgnore")
+    private String kmersToIgnoreFilename;
+
+    @Override
+    public boolean requiresReads()
+    {
+        return true;
+    }
+
+    @Override
+    protected void runTool( final JavaSparkContext ctx ) {
+        if ( getHeaderForReads().getSortOrder() != SAMFileHeader.SortOrder.coordinate ) {
+            throw new GATKException("The reads must be coordinate sorted.");
+        }
+
+        final Locations locations =
+                new Locations(evidenceDir, intervalFile, qNamesMappedFile, kmerFile, qNamesAssemblyFile);
+        final PipelineOptions pipelineOptions = getAuthenticatedGCSOptions();
+        final SAMFileHeader header = getHeaderForReads();
+        final JavaRDD<GATKRead> reads = getUnfilteredReads();
```

Comment:
Sadly, the filtering varies slightly between passes, and must vary for correct behavior.  For example, when producing the FASTQ from a set of template names we want to make sure that we always get both members of each pair exactly once.  So we must filter down to primary lines to avoid dups, but don't want any quality filters or we might exclude a poor quality mate.  On the other hand, when we're looking for reads to support a breakpoint hypothesis secondary and supplementary alignment records are useful, but dups, vendor quality check failures, and unmapped reads are not.  Completely disjoint set of filtering criteria.
Nonetheless, I'll make an effort to organize the filtering a little better and avoid repetition.